### PR TITLE
Migrate all main/v5 code to OSS repo

### PR DESCRIPTION
**Pull request by stippi2**
The process was:
- Remove all module directories from OSS repo in one commit
- Add all module directories from main/v5 branch (Oct 02 ~0900) to OSS repo in one commit authored by `cloudsdk@sap.com`
- Update files in `scripts` with their version from `scripts/python` from main/v5
- Look closely at the diff of all pom.xml files & restore changes from OSS Repo:
  - Remove cx-server related stuff from archetypes resources
  - `modules-bom/pom.xml`
    - Reformat
    - Restore `<build>` section with plugins from the previous OSS version of the file
  - `bom/pom.xml`:
    - Reformat
    - Add `<build>` section with plugins (see above)
  - Restore root POM Javadoc changes
  - Restore root POM Jacoco changes (:warning: please check)
  - Add property `project.build.sourceEncoding` in `bom/pom.xml` and `modules-bom/pom.xml`, required for Code Formatting check to pass. The OSS version of `modules-bom/pom.xml` previously configured the `encoding` property of the formatter plugin in the build section. This should be an equivalent change.
  - Adjust the allowed "high" findings count for PMD, SpotBugs and FindBugs. (:warning: needs proper fix via correct exclusions)

:warning: Please don't squash merge this one, in order to preserve authorship of `cloudsdk@sap.com` for all source code.

### CODE REVIEW

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/ComplexDestinationPropertyFactory.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+            } else {
+                fallback = AuthenticationType.NO_AUTHENTICATION;
+            }
+
+            log
+                .debug(
+                    "No valid JSON primitive '{}' or '{}' defined. Falling back to {}.",
+                    DestinationProperty.AUTH_TYPE,
+                    DestinationProperty.AUTH_TYPE_FALLBACK,
+                    fallback);
```
[the remark this answers is not in the document]
Comment (stippi2) [reply cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/ComplexDestinationPropertyFactory.java:224]: I need more context in order to understand this comment.

**Comment by CharlesDuboisSAP** [comment]
- The bom/pom.xml shouldn't have the extra plugins and the extra `<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>`.
- The modules-bom/pom.xml has an extra `<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>` that can be removed.
- (Minor) Every pom but the two boms are formatted with tabs, those two should be consistent
- The use of `http` or `https` `://www.apache.org/licenses/LICENSE-2.0.txt` is inconsistent throughout the repo
- What is the new file `.adr-dir`.
- Remove the commented jacoco config in the main parent pom
- The PMD plugin does not find all 701 warnings, only 14
- Findbugs found 800 bugs -> did it never work before? If yes then adjust thresholds
- Discrepancy in the Checkstyle results, also adjusts thresholds after
- Where can we find coverageThresholds? Also adjust them if the report can be found
- line missing in intellij_java_style
- Why was the config removed from the scp-cf-spring archetype?
- Why was the Setup java step added to the .github/workflow.build.yml?
- Remove the extra .gitignore files (our fault)
- scripts/common/release_audience.py has an extra "Internal" audience, is this intentional?
- Unintentional change line 40 in scripts/create_module_invetory_file.py

**Comment by stippi2** [comment]
> * The bom/pom.xml shouldn't have the extra plugins and the extra `<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>`.
> * The modules-bom/pom.xml has an extra `<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>` that can be removed.

The "Code Formatting" job in the pipeline fails if the `encoding` property of the `formatter-maven-plugin` is not defined. Since those two POMs don't inherit from the sdk-parent POM, the property is not configured. Maybe the POMs should be excluded from the formatting checks, but that doesn't seem plausible to me. Should the encoding be specified some other way?

**Comment by stippi2** [comment]
> * Why was the Setup java step added to the .github/workflow.build.yml?

Because one of the modules requires Java 17 as a minimum, which doesn't seem to the default on the GitHub runner when this job is executed. As a result, if we don't pre-configure Java, the CodeQL scan fails for this module.

**Comment by stippi2** [comment]
> * scripts/common/release_audience.py has an extra "Internal" audience, is this intentional?

Yes, since at least one of the modules has the audience property with this value, and generating the `modules_inventory.json` file then fails. We do generate this faile as part of the pipeline and validate that it doesn't result in changes.

**Comment by stippi2** [comment]
> * Where can we find coverageThresholds? Also adjust them if the report can be found

Can be found [here](https://github.com/SAP/cloud-sdk-java/blob/91316a04b56d9aab0098e57b015633e734f10c64/.pipeline/config.yml#L13) and there is a TODO for adjusting it after migrating the code. Should I adjust within this PR?

**Comment by CharlesDuboisSAP** [comment]
> > * scripts/common/release_audience.py has an extra "Internal" audience, is this intentional?
> 
> Yes, since at least one of the modules has the audience property with this value, and generating the `modules_inventory.json` file then fails. We do generate this faile as part of the pipeline and validate that it doesn't result in changes.

I will set the modules to Public and remove the audience type Internal.

File: .pipeline/config.yml [hunk 2]
```diff
@@ -26,14 +26,14 @@ stages:
...
     findbugs:
-      high: '0'
+      high: '10'
       normal: '-1'
       low: '-1'
     pmd:
-      high: '0'
+      high: '10'
       normal: '-1'
       low: '-1'
```
Comment (CharlesDuboisSAP) [inline .pipeline/config.yml:39]: > > * Where can we find coverageThresholds? Also adjust them if the report can be found
> 
> Can be found [here](https://github.com/SAP/cloud-sdk-java/blob/91316a04b56d9aab0098e57b015633e734f10c64/.pipeline/config.yml#L13) and there is a TODO for adjusting it after migrating the code. Should I adjust within this PR?

Yes we should adjust the thresholds in this PR.
@adumontb @KavithaSiva do you think we should investigate the difference in amount of findings?

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/ComplexDestinationPropertyFactory.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (CharlesDuboisSAP) [reply cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/ComplexDestinationPropertyFactory.java:224]: I already closed them as a false positive with my comment in explanation

File: .pipeline/config.yml (hunk 2 above)
Comment (adumontb) [reply .pipeline/config.yml:39]: Hi!
I don't know why I was added as a reviewer on this PR. I'm a technical writer working on Commerce Cloud. Could you please remove me? Thank you!

**Comment by stippi2** [comment]
Could the PMD and CheckStyle differences be explained by the different rulesets (see root-POM changes)?

**Comment by CharlesDuboisSAP** [comment]
LGTM, but let's wait for someone else's review.

**Comment by stippi2** [comment]
Let's get the excludes working for Jacoco coverage first...

**Review by newtork — APPROVED** [review]
yolo

**Pull request merged**